pyrene/src/components/DropdownButton/DropdownButton.tsx

```diff
@@ -1,96 +1,94 @@
-import React, { useState } from 'react';
-import PropTypes from 'prop-types';
+import React, { FunctionComponent, useState } from 'react';
 import clsx from 'clsx';
 import Popover from '../Popover/Popover';
 import OptionsList from './OptionsList';
-import './dropdownButton.css';
 import Loader from '../Loader/Loader';
+import './dropdownButton.css';
 
 /**
  * Dropdown Buttons are used primarily to group actions of a common topic.
  * They have the same appearance and behaviour as the regular Button.
  */
-const DropdownButton = (props) => {
 
-  const [state, setState] = useState({
-    displayActions: false,
-  });
+export interface DropdownButtonProps {
+  /**
+   * Array of action objects holding a Label, Icon and its associated Javascript event handler.
+   */
+  actions?: Array<{
+    label: string,
+    onClick: () => void,
+  }>,
+  /**
+   * Sets the alignment of the dropdown relative to the button.
+   */
+  align?: 'start' | 'end',
+  /**
+   * Disables any interaction with the component.
+   */
+  disabled?: boolean,
+  /**
+   * Adds an icon to the element.
+   */
+  icon?: string,
+  /**
+   * Sets the label displayed to the user.
+   */
+  label: string,
+  /**
+   * Disables the component and displays a loader inside of it.
+   */
+  loading?: boolean,
+}
 
-  const onClose = () => setState({ displayActions: false });
+const DropdownButton: FunctionComponent<DropdownButtonProps> = ({
+  actions = [{ label: 'tbd', onClick: () => null }],
+  align = 'start',
+  disabled = false,
+  icon = '',
+  label,
+  loading = false,
+}: DropdownButtonProps) => {
+  const [displayActions, setDisplayActions] = useState(false);
+  const onClose = () => setDisplayActions(false);
 
   return (
     <Popover
-      displayPopover={state.displayActions}
+      displayPopover={displayActions}
       onClickOutside={onClose}
-      renderPopoverContent={() => <OptionsList actions={props.actions} onClick={onClose} />}
+      renderPopoverContent={() => <OptionsList actions={actions} onClick={onClose} />}
       preferredPosition={['bottom']}
       distanceToTarget={4}
-      align={props.align}
+      align={align}
     >
       <div className="buttonContainer">
         <button
           type="submit"
           styleName={
             clsx('button',
-              { disabled: props.disabled },
-              { loading: props.loading },
-              { openedDropdown: state.displayActions })
+              { disabled: disabled },
+              { loading: loading },
+              { openedDropdown: displayActions })
           }
-          disabled={props.disabled || props.loading}
-          onClick={() => setState({ displayActions: !state.displayActions })}
+          disabled={disabled || loading}
+          onClick={() => setDisplayActions((prevDisplayActions) => !prevDisplayActions)}
```

Comment:
> Wouldn't this also work?
> `onClick={() => setDisplayActions(!prevDisplayActions)}`

If the update depends on the current, it s [recommended](https://reactjs.org/docs/hooks-reference.html#functional-updates) to use a callback (since it s asynchronous).